web/src/ui_init.js

```diff
@@ -735,9 +735,13 @@ export function initialize_everything(state_data) {
         on_mark_pm_as_read: unread_ops.mark_pm_as_read,
         on_mark_topic_as_read: unread_ops.mark_topic_as_read,
         maybe_load_older_messages() {
+            recent_view_ui.set_backfill_in_progress(true);
             message_fetch.maybe_load_older_messages({
                 msg_list_data: all_messages_data,
                 recent_view: true,
+                // To have a hard anchor on our target of first unread message id,
+                // we pass it from here, otherwise it might get updated and lead to confusion.
+                first_unread_message_id: unread.get_all_msg_ids()[0],
```

Comment:
Added a comment noting a bug in this logic. This is probably a blocker for sending it to production. We can probably address this with a function along the lines of `with_first_message_id` to just ask the server for the number.

``` diff
diff --git a/web/src/ui_init.js b/web/src/ui_init.js
index f400b8d82d..f334ee0c14 100644
--- a/web/src/ui_init.js
+++ b/web/src/ui_init.js
@@ -740,6 +740,9 @@ export function initialize_everything(state_data) {
                 recent_view: true,
                 // To have a hard anchor on our target of first unread message id,
                 // we pass it from here, otherwise it might get updated and lead to confusion.
+                //
+                // TODO: Ideally, muted unread messages would not
+                // count for this purpose.
                 first_unread_message_id: unread.get_all_msg_ids()[0],
             });
         },
```

Reply:
Opened #30060